### ISSUE TITLE
Internal Server Error when logging in through GitHub

### ISSUE THREAD

**Issue by th0mas** [dwyl/auth#124]
When logging in through GitHub or creating an email account I get an internal server error:

![image](https://user-images.githubusercontent.com/9089056/93493140-7ddbdb00-f903-11ea-9df1-3e0b6137809a.png)

This only happens when trying to authenticate with my local development application and I cant replicate this anywhere else, it works on the Heroku hosted version...

I've reset my API keys and still get the error.

**Comment by th0mas** [comment]
This seems to be a URL formatting bug as the callback parameter string is

```
code=e772154f8879f954b479&state=http%3A%2F%2Flocalhost%3A4000%2F%3Fauth_client_id%3D2cfxNawpSpsRazGQEHQ5eoRB2nXtEMZve8LBZrQ5e5JCbKk68sT
```

Which is not a valid URL...

**Comment by th0mas** [comment]
From looking at the Heroku logs were not correctly deleting old API keys, so when I've "Reset" them, it causes an error when multiple API keys are pulled from the database.

```
2020-09-17T16:08:26.645265+00:00 app[web.1]: ** (exit) an exception was raised:
2020-09-17T16:08:26.645266+00:00 app[web.1]:     ** (Ecto.MultipleResultsError) expected at most one result but got 8 in query:
2020-09-17T16:08:26.645267+00:00 app[web.1]: 
2020-09-17T16:08:26.645267+00:00 app[web.1]: from a0 in Auth.Apikey,
2020-09-17T16:08:26.645268+00:00 app[web.1]:   where: a0.app_id == ^6
```

**Comment by nelsonic** [comment]
Ah ... that's _very_ interesting. 💭 
Looks like we need to add the "status_id != 6" ("deleted") to the where clause.
That should be an easy fix.

**Comment by nelsonic** [comment]
@th0mas want me to fix it as part of my current PR (_with a regression test_) or are you on it? ❓

**Comment by th0mas** [comment]
@nelsonic that would be great

**Comment by nelsonic** [comment]
@th0mas this should be fixed on Heroku and included in #123 :shipit: 
Please confirm (_and close this issue_) when you're back at your desk this week. 👍 
Thanks! ☀️

**Comment by th0mas** [comment]
Yeah this works now, cheers
